packages/plugin-core/src/test/suite-integ/NoteLookupCommand.test.ts

```diff
@@ -749,6 +752,433 @@ suite("NoteLookupCommand", function () {
+            fname: "template.ch2",
+            vault: vaults[1],
+          });
+        },
+        postSetupHook: async ({ wsRoot, vaults }) => {
+          // Schema is in vault1
+          const vault = vaults[0];
+          await NoteTestUtilsV4.createSchema({
+            fname: "food",
+            wsRoot,
+            vault,
+            modifier: (schema) => {
+              const schemas = [
+                SchemaUtils.createFromSchemaOpts({
+                  id: "food",
+                  parent: "root",
+                  fname: "food",
+                  children: ["ch2"],
+                  vault,
+                }),
+                SchemaUtils.createFromSchemaRaw({
+                  id: "ch2",
+                  template: {
+                    id: "template.ch2",
+                    type: "note",
+                  },
+                  namespace: true,
+                  vault,
+                }),
+              ];
+              schemas.map((s) => {
+                schema.schemas[s.id] = s;
+              });
+              return schema;
+            },
+          });
+        },
+      },
+      () => {
+        test("THEN template body gets applied to new note", async () => {
+          const cmd = new NoteLookupCommand();
+          await cmd.run({
+            initialValue: "food.ch2",
+            noConfirm: true,
+          });
+          const { engine, vaults } = ExtensionProvider.getDWorkspace();
+
+          const newNote = NoteUtils.getNoteByFnameFromEngine({
+            fname: "food.ch2",
+            engine,
+            vault: vaults[0],
+          });
+          expect(_.trim(newNote?.body)).toEqual("food ch2 template");
+        });
+      }
+    );
+
+    describeMultiWS(
+      "WHEN schema template references to a template note that lies in a different vault using xvault notation",
+      {
+        ctx,
+        preSetupHook: async ({ wsRoot, vaults }) => {
+          await ENGINE_HOOKS_MULTI.setupBasicMulti({ wsRoot, vaults });
+          // Template is in vault2
+          await NoteTestUtilsV4.createNote({
+            wsRoot,
+            body: "food ch2 template",
+            fname: "template.ch2",
+            vault: vaults[1],
+          });
+        },
+        postSetupHook: async ({ wsRoot, vaults }) => {
+          // Schema is in vault1
+          const vault = vaults[0];
+          await NoteTestUtilsV4.createSchema({
+            fname: "food",
+            wsRoot,
+            vault,
+            modifier: (schema) => {
+              const schemas = [
+                SchemaUtils.createFromSchemaOpts({
+                  id: "food",
+                  parent: "root",
+                  fname: "food",
+                  children: ["ch2"],
+                  vault,
+                }),
+                SchemaUtils.createFromSchemaRaw({
+                  id: "ch2",
+                  template: {
+                    id: `dendron://${vaults[1].fsPath}/template.ch2`,
```

Comment:
This should use `VaultUtils.getName` instead

Reply:
fixed